### PR TITLE
pump: Refactor storage to make it easier to extend

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

The following two pieces of code can not be easily reused:

1. The code for writing a batch of requests to LevelDB
2. The code for scanning vlog for requests

### What is changed and how it works?

Extract functions for the two pieces of code.
Also make sure goroutines are used in the test helper function `populateBinlog`.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

File: pump/storage/storage.go [hunk 1]
```diff
@@ -1122,3 +1072,39 @@ func openMetadataDB(kvDir string, cf *KVConfig) (*leveldb.DB, error) {
 
 	return leveldb.OpenFile(kvDir, &opt)
 }
+
+func (a *Append) writeBatchToKV(bufReqs []*request) error {
+	var batch leveldb.Batch
+	var lastPointer []byte
+	for _, req := range bufReqs {
+		log.Info("write request to kv", zap.Reflect("request", req))
+		var err error
```
Comment (GregoryIan) [inline pump/storage/storage.go]: remove it

File: pump/storage/storage_test.go [hunk 2]
```diff
@@ -269,7 +269,7 @@ func populateBinlog(b Log, append *Append, prewriteValueSize int, binlogNum int3
 
 	for i := 0; i < goNum; i++ {
 		wg.Add(1)
-		func() {
+		go func() {
```
Comment (july2993) [inline pump/storage/storage_test.go]: 🤪

**Comment by GregoryIan** [comment]
Good job! LGTM

File: pump/storage/storage_test.go (hunk 2 above)
Comment (suzaku) [reply pump/storage/storage_test.go]: 🤣

File: pump/storage/storage.go [hunk 3]
```diff
@@ -1122,3 +1072,39 @@ func openMetadataDB(kvDir string, cf *KVConfig) (*leveldb.DB, error) {
 
 	return leveldb.OpenFile(kvDir, &opt)
 }
+
+func (a *Append) writeBatchToKV(bufReqs []*request) error {
+	var batch leveldb.Batch
+	var lastPointer []byte
+	for _, req := range bufReqs {
+		log.Info("write request to kv", zap.Reflect("request", req))
```
Comment (july2993) [inline pump/storage/storage.go]: why chang to Info level, will too many log

File: pump/storage/storage.go (hunk 1 above)
Comment (suzaku) [reply pump/storage/storage.go]: Done.

**Pull request merged**